### ISSUE TITLE
Allow for duplicates in index?

### ISSUE THREAD

**Issue by hagenw** [audeering/audformat#86]
At the moment we have the problem, that we don't cover some use cases with our available tables in `audformat`.
If you have as input the comparison between two files, or a file combined with another column as combination and a third with the result, we can not handle it as we do not allow for duplicate index entries.

One example would be a table used for verification experiments:

```
                                                   verification file  same speaker
file                                                                              
wav/id10270/x6uYqmx31kE/00001.wav  wav/id10300/ize_eiCFEg0/00003.wav         False
wav/id10270/x6uYqmx31kE/00001.wav  wav/id10270/GWXujl-xAVM/00017.wav          True
wav/id10270/x6uYqmx31kE/00001.wav  wav/id10273/0OCW1HUxZyg/00001.wav         False
wav/id10270/x6uYqmx31kE/00001.wav  wav/id10270/8jEAjG6SegY/00022.wav          True
```

**Comment by hagenw** [comment]
We decided now to extend the index and not go with duplicated entries for the speaker verification use case.

I think we should change the tests for `audformat` conform tables, and simply require that the first column of the index needs to be `file`. If the 2nd and 3rd column are `start` and `end` it is a segmented table, otherwise a filewise table, but the users can then add as many table columns as they want.

**Comment by frankenjoe** [comment]
Allowing for more index columns might be quite tricky, see #139 

Another solution for the "comparison problem" you mention here is to stick with a filewise index, but add support columsn with tuples:

```
                                                   verification files
file                                                                              
wav/id10270/x6uYqmx31kE/00001.wav  (wav/id10270/GWXujl-xAVM/00017.wav, wav/id10270/8jEAjG6SegY/00022.wav)
...

**Comment by hagenw** [comment]
> Allowing for more index columns might be quite tricky, see #139
> 
> Another solution for the "comparison problem" you mention here is to stick with a filewise index, but support columns with tuples:
> 
> ```
>                                    verification files
> file                                                                              
> wav/id10270/x6uYqmx31kE/00001.wav  (wav/id10270/GWXujl-xAVM/00017.wav, wav/id10270/8jEAjG6SegY/00022.wav)
> ...
> ```

The problem is that in reality you quickly need a segmented index for comparison and there we have the problem that our `timedelta` entries in `start` and `end` behave differently if they are stored in an index or as a tuple in a column. I tried this already and it was a nightmare ;).

**Comment by hagenw** [comment]
Yes, I have seen this already in official speaker verification benchmarks and I also needed it when using files from projectsmile.

**Comment by frankenjoe** [comment]
Ok, another solution would be to allow multiples in the index. I guess we had reasons to forbid it at some point, but since I don't really see #139 to work, we should probably consider it again.

**Comment by frankenjoe** [comment]
I was thinking of:
 
```                  verification-file verification-start verification-end
file start end  ...
```

**Comment by hagenw** [comment]
> I was thinking of:
> 
> ```
>                 verification-file verification-start verification-end
> file start end  ...
> ```

Yes, in principle this is fine, but as stated above I did not get it to work due to `timedelta` values automatically converted to `datetime` or whatever values when loading them from a data column instead of the index.

**Comment by hagenw** [comment]
But maybe we should give it another try as I also see big problems with https://github.com/audeering/audformat/pull/139.

**Comment by frankenjoe** [comment]
> `timedelta` values automatically converted to `datetime` or whatever values when loading them from a data column instead of the index

ok, that we should fix anyway cause it basically means that the `time` datatype a user can select as scheme for column is not working.

**Comment by hagenw** [comment]
> > `timedelta` values automatically converted to `datetime` or whatever values when loading them from a data column instead of the index
> 
> ok, that we should fix anyway cause it basically means that the `time` datatype a user can select as scheme for column is not working.

If we are lucky it happened only when I stored them as a tuple `(file, start, end)`, but I'm not sure about it, let's see.

**Comment by hagenw** [comment]
I started working on updating `set()` in https://github.com/audeering/audformat/pull/177

**Comment by frankenjoe** [comment]
Ok good point. But yes, I think it's ok if we leave this to the responsibility of the user.

**Comment by frankenjoe** [comment]
Supporting index-less table could generally be a useful feature to store any kind of data / labels that is not well covered by filewise and segmented tables.

**Comment by hagenw** [comment]
Yes, sounds like a good idea. So let's target to introduce an index-less table, see https://github.com/audeering/audformat/issues/178
